### PR TITLE
Add playbooks and roles to launch VMs on ovirt/rhv

### PR DESCRIPTION
**Pull request by fabianvf**
Relies on ovirt/RHV 4.0.6

### CODE REVIEW

File: roles/subscription/tasks/main.yml [hunk 1]
```diff
@@ -7,4 +7,14 @@
...
 - name: enable repos
-  command: subscription-manager repos --enable {{ repositories | join(' --enable ')}}
+  command: subscription-manager repos --enable {{ repositories | join(' --enable ')}} --enable jb-eap-7-for-rhel-7-server-rpms
+
+- name: create rhev repo
+  copy:
+    content: |
+      [rhev]
+      name=rhev
+      baseurl=http://bob.eng.lab.tlv.redhat.com/builds/latest_4.0/el7Server/
```
Comment (mykaul) [inline roles/subscription/tasks/main.yml]: We don't usually have internal URLs available externally.

File: roles/subscription/tasks/main.yml [hunk 2]
```diff
@@ -7,4 +7,14 @@
   command: subscription-manager repos --disable=*
 
 - name: enable repos
-  command: subscription-manager repos --enable {{ repositories | join(' --enable ')}}
+  command: subscription-manager repos --enable {{ repositories | join(' --enable ')}} --enable jb-eap-7-for-rhel-7-server-rpms
+
+- name: create rhev repo
+  copy:
+    content: |
+      [rhev]
```
Comment (mykaul) [inline roles/subscription/tasks/main.yml]: rhev -> rhv

File: roles/launch_vms/files/upload_image [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+    }
+
+    # Extract the URI, port, and path from the transfer's proxy_url.
+    url = urlparse(transfer.proxy_url)
+
+    # Create a new instance of the connection.
+    proxy_connection = HTTPSConnection(
+        url.hostname,
+        url.port,
+        context=ssl.SSLContext(ssl.PROTOCOL_SSLv23)
```
Comment (machacekondra) [inline roles/launch_vms/files/upload_image]: Why SSLv23?

File: roles/launch_vms/files/upload_image [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+                url.path,
+                disk.read(chunk_size),
+                headers=upload_headers
+            )
+            r = proxy_connection.getresponse()
+            progress = pos / float(size)
+            print(r.status, r.reason, "Progress: ", "{:.0%}".format(progress))
+            pos += chunk_size
+
+    transfer_service.finalize()
```
Comment (machacekondra) [inline roles/launch_vms/files/upload_image]: You should check if it succed, but this operation support ovirt_disks modules so no need to do it here.

**Comment by machacekondra** [comment]
I think it would be better to copy the content of https://github.com/ansible/ansible/tree/devel/lib/ansible/modules/cloud/ovirt into library directory and rewrite the playbooks to use it, so we can improve the modules, instead of redudant work.

File: roles/launch_vms/files/upload_image (hunk 3 above)
Comment (fabianvf) [inline roles/launch_vms/files/upload_image]: That I'm not sure, I grabbed this line from the official docs

File: roles/launch_vms/files/upload_image (hunk 4 above)
Comment (fabianvf) [inline roles/launch_vms/files/upload_image]: Good to know, I was not aware

**Comment by fabianvf** [comment]
Interesting, that hadn't occurred to me. I'll bring that up with my team, it would be nice to not be waiting for the correct ansible release.

File: roles/launch_vms/files/upload_image [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+                )
+            ]
+        )
+    )
+
+
+def wait_for_disk(disks_service, disk):
+    disk_service = disks_service.disk_service(disk.id)
+    while disk.status != types.DiskStatus.OK:
+        time.sleep(5)
```
Comment (michalskrivanek) [inline roles/launch_vms/files/upload_image]: every upload is going to be relatively long operation and our API is quite slow...15s sounds better to me

**Comment by fabianvf** [comment]
I will make the changes suggested in the comments, just putting this into master so we can get it into the next compose.

**Pull request merged**